packages/react/src/components/UIShell/__tests__/SideNavMenu-test.js

```diff
@@ -98,4 +102,118 @@ describe('SideNavMenu', () => {
     );
     expect(ref).toHaveBeenCalledWith(screen.getByRole('button'));
   });
+
+  it('should return true if the child is a valid React element, and instance of Array and has isActive and aria-current props', () => {
+    const child = [
+      <SideNavMenuItem isActive={true} aria-current="page">
+        <SideNavMenuItem isActive={false} aria-current="page">
+          a
+        </SideNavMenuItem>
+      </SideNavMenuItem>,
+      <SideNavMenuItem isActive={true} aria-current="page">
+        b
+      </SideNavMenuItem>,
+    ];
+    expect(hasActiveDescendant(child)).toBe(true);
+  });
```

Comment:
Rather than test the internal implementation like this - how could we reframe the tests from the perspective of someone using the component?

hasActiveDescendant is only called in one place under certain circumstances
https://github.com/carbon-design-system/carbon/blob/061970723f64d1376ea28109328c1aec1f192ecc/packages/react/src/components/UIShell/SideNavMenu.tsx#L91-L92

I think this would mean that to test this from a user-interaction viewpoint, each test could `render` the component with a different type of children which would get `hasActiveDescendant` to be called, and then the assertion would be `toHaveClass('${prefix}--side-nav__item--active')`.